### PR TITLE
Publish PRs for Source-Build Test Results

### PR DESCRIPTION
**Pull request by ellahathaway**
Transfer of https://github.com/dotnet/installer/pull/19434 into SDK

Automates the creation of PRs given the updated test files produced by the Source Build sdk content and license scan tests.

### CODE REVIEW

**Review by mthalman — APPROVED** [review]

File: src/SourceBuild/content/eng/pipelines/templates/jobs/sdk-diff-tests.yml [hunk 1]
```diff
@@ -165,3 +169,12 @@ jobs:
...
       testRunTitle: $(Agent.JobName)
+
+  - ${{ if and(eq(parameters.publishTestResultsPr, 'true'), or(eq(variables['Build.SourceBranch'], 'refs/heads/main'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release'))) }}:
+      - template: ../steps/create-baseline-update-pr.yml
+        parameters:
+          pipeline: sdk
+          repo: dotnet/sdk
+          originalTestResultsPath: src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets
+          updatedTestResultsPath: $(Build.StagingDirectory)/BuildLogs
+          pullRequestTitle: Update SDK Diff Tests Baselines and Exclusions
```
Comment (MichaelSimons) [inline src/SourceBuild/content/eng/pipelines/templates/jobs/sdk-diff-tests.yml:180]: I think source-build should be included in the title.  There are UB SDK diff tests as well.

File: src/SourceBuild/content/eng/pipelines/templates/steps/create-baseline-update-pr.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+parameters:
+# The pipeline that is being run
+# Used to determine the correct baseline update tool to run
+# Currently only supports "sdk" and "license"
+- name: pipeline
+  type: string
+
+- name: repo
```
Comment (MichaelSimons) [inline src/SourceBuild/content/eng/pipelines/templates/steps/create-baseline-update-pr.yml:8]: A comment would be helpful here indicating the expected format.  e.g. This is not a GitHub repo URL

File: src/SourceBuild/content/eng/pipelines/templates/steps/create-baseline-update-pr.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  type: string
+
+- name: originalTestResultsPath
```
Comment (MichaelSimons) [inline src/SourceBuild/content/eng/pipelines/templates/steps/create-baseline-update-pr.yml:11]: The term TestResults is a little confusing to me as we have a TestResults subfolder in the artifacts directory.  The meaning here is different.  How about simply naming them originalFilesDirectory and updateFilesDirectory?

File: src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/PRCreator.cs [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+    {
+        // Create a new GitHub client
+        _client = new GitHubClient(new ProductHeaderValue(System.Reflection.Assembly.GetExecutingAssembly().GetName().Name));
+        var authToken = new Credentials(gitHubToken);
+        _client.Credentials = authToken;
+        _repoOwner = repo.Split('/')[0];
+        _repoName = repo.Split('/')[1];
+    }
+
+    private static readonly string DefaultLicenseBaselineContent = "{\n  \"files\": []\n}";
```
Comment (MichaelSimons) [inline src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/PRCreator.cs:29]: Why static readonly vs a const?  Regardless this feels better places next to the consts

**Review by MichaelSimons — COMMENTED** [review]
I just took a cursory look and added a few minor comments.  I will rely on Matt for an in-depth technical review.

File: src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/Directory.Build.props [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<Project>
+
+  <!--
+    Do not import the Arcade SDK as this would introduce a prebuilt as this project builds
+    before the custom sdk resolver is available in source-only mode.
+  -->
```
Comment (ViktorHofer) [inline src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/Directory.Build.props:6]: Does this library build as part of the VMR orchestrator build? If not then I would remove this comment and file completely. I think it should be fine to use the Arcade SDK here.

File: src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/PRCreator.cs (hunk 4 above)
Comment (ellahathaway) [reply src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/PRCreator.cs:29]: Whoops. Forgot to update this. Will fix.

File: src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/Directory.Build.props (hunk 5 above)
Comment (ellahathaway) [reply src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/Directory.Build.props:6]: Good point, this shouldn't get built as part of the VMR. I'll remove this file.

**Review by mthalman — APPROVED** [review]

**Pull request merged**